### PR TITLE
Greyscale ID Department Colors (+ Rainbow ID Card)

### PR DESCRIPTION
**Pull request by unit0016**
## About The Pull Request
Greyscales ID department colors. Consequently, all color differences are now consistent.
Adds assigned icon states for gold and silver ID cards, updates radial.dmi to accommodate for the former.
As a result of greyscaling the clown's ID (Pour one out for a real one, rainbow trim), I've implemented the ""unused"" rainbow ID card for the clown ONLY. It's functionally identical to a grey ID besides looking... different.
image

TODO:

- [x] ~~Fix RPG Titles not updating trim state properly (????)~~ This is still broken, but isn't actually my fault, best I can tell...? Cool?
- [x] Allow people to change ID card colors in-game (Crayon?) **I HATE CRAYONCODE I HATE CRAYONCODE**

## Why It's Good For The Game
Shared pixels suck. Admins can now VV to color any ID however they want. Players now have the power to recolor any ID's department. ~~Return to green medbay.~~

## Changelog

:cl:
refactor: ID Card department colors are now greyscaled. Say whuh!?
/:cl:

### CODE REVIEW

**Comment by VladinXXV** [comment]
the rainbow id card isn't unused iirc, you can get it using the cargo drones

**Comment by unit0016** [comment]
> i personally think it should remain somethin the clown has to work for.

Unless we want snowflake code *just* for the clown's trim giving them a rainbow card is better than trying to re-implement the rainbow trim.

File: code/datums/id_trim/_id_trim.dm [hunk 1]
```diff
@@ -4,6 +4,12 @@
 	var/trim_icon = 'icons/obj/card.dmi'
 	/// Icon state for this trim. Overlayed on advanced ID cards.
 	var/trim_state
+	/// Department color for this trim. Displayed in the box under the trim_state.
+	var/department_color = "#6E6E6E"
```
Comment (Mothblocks) [inline code/datums/id_trim/_id_trim.dm:8]: Use/make color defines

Comment (Mothblocks) [reply code/datums/id_trim/_id_trim.dm:8]: Here and everywhere else

File: code/datums/id_trim/centcom.dm [hunk 2]
```diff
@@ -175,7 +185,8 @@
 /// Trim for generic ERT clowns. No universal ID card changing access.
 /datum/id_trim/centcom/ert/clown
 	assignment = JOB_ERT_CLOWN
-	trim_state = "trim_ert_entertainment"
+	trim_state = "trim_clown"
+	subdepartment_color = "#FF00FF" // RIP Rainbow trim, a real one
```
Comment (Mothblocks) [inline code/datums/id_trim/centcom.dm:189]:
```suggestion
	subdepartment_color = "#FF00FF"
```

Comments are for today

File: code/game/objects/items/cards_ids.dm [hunk 3]
```diff
@@ -901,6 +908,21 @@
...
+	. = ..()
+	if(istype(W, /obj/item/toy/crayon))
+		var/obj/item/toy/crayon/our_crayon = W
+		if(tgui_alert(usr, "Recolor Department or Subdepartment?", "Recoloring ID...", list("Department", "Subdepartment")) == "Department")
+			if(do_after(user, 2 SECONDS)) // Doesn't technically require a spraycan's cap to be off but shhh
+				src.department_color_override = our_crayon.paint_color
+				balloon_alert(user, "Recolored!")
+			else return
+		else if(do_after(user, 1 SECONDS))
+			src.subdepartment_color_override = our_crayon.paint_color
```
Comment (Mothblocks) [inline code/game/objects/items/cards_ids.dm:922]:
```suggestion
			subdepartment_color_override = our_crayon.paint_color
```

File: code/game/objects/items/cards_ids.dm [hunk 4]
```diff
@@ -901,6 +908,21 @@
...
+
+/obj/item/card/id/advanced/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	if(istype(W, /obj/item/toy/crayon))
+		var/obj/item/toy/crayon/our_crayon = W
+		if(tgui_alert(usr, "Recolor Department or Subdepartment?", "Recoloring ID...", list("Department", "Subdepartment")) == "Department")
+			if(do_after(user, 2 SECONDS)) // Doesn't technically require a spraycan's cap to be off but shhh
+				src.department_color_override = our_crayon.paint_color
+				balloon_alert(user, "Recolored!")
+			else return
```
Comment (Mothblocks) [inline code/game/objects/items/cards_ids.dm:920]: Make this an early return instead

File: code/game/objects/items/cards_ids.dm [hunk 5]
```diff
@@ -901,6 +908,21 @@
...
 	return ..()
 
+
+/obj/item/card/id/advanced/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	if(istype(W, /obj/item/toy/crayon))
+		var/obj/item/toy/crayon/our_crayon = W
+		if(tgui_alert(usr, "Recolor Department or Subdepartment?", "Recoloring ID...", list("Department", "Subdepartment")) == "Department")
+			if(do_after(user, 2 SECONDS)) // Doesn't technically require a spraycan's cap to be off but shhh
+				src.department_color_override = our_crayon.paint_color
```
Comment (Mothblocks) [inline code/game/objects/items/cards_ids.dm:918]:
```suggestion
				department_color_override = our_crayon.paint_color
```

File: code/game/objects/items/cards_ids.dm [hunk 6]
```diff
@@ -901,6 +908,21 @@
...
+				balloon_alert(user, "Recolored!")
```
Comment (Mothblocks) [inline code/game/objects/items/cards_ids.dm:919]:
```suggestion
				balloon_alert(user, "recolored")
```

File: code/game/objects/items/cards_ids.dm [hunk 7]
```diff
@@ -901,6 +908,21 @@
...
+			else return
+		else if(do_after(user, 1 SECONDS))
+			src.subdepartment_color_override = our_crayon.paint_color
+			balloon_alert(user, "Recolored!")
```
Comment (Mothblocks) [inline code/game/objects/items/cards_ids.dm:923]:
```suggestion
			balloon_alert(user, "recolored")
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/datums/id_trim/_id_trim.dm (hunk 1 above)
Comment (unit0016) [reply code/datums/id_trim/_id_trim.dm]: Switched out for defines.

File: code/game/objects/items/cards_ids.dm [hunk 8]
```diff
@@ -901,6 +908,21 @@
...
+
+/obj/item/card/id/advanced/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	if(istype(W, /obj/item/toy/crayon))
+		var/obj/item/toy/crayon/our_crayon = W
+		if(tgui_alert(usr, "Recolor Department or Subdepartment?", "Recoloring ID...", list("Department", "Subdepartment")) == "Department")
+			if(do_after(user, 2 SECONDS)) // Doesn't technically require a spraycan's cap to be off but shhh
+				department_color_override = our_crayon.paint_color
+				balloon_alert(user, "recolored")
+			return
```
Comment (Mothblocks) [inline code/game/objects/items/cards_ids.dm:920]: This won't call update_icon(). That is not what I meant by early-return.

I meant:

```dm
if (!do_after(user, 2 SECONDS))
    return

// do stuff here

Comment (unit0016) [reply code/game/objects/items/cards_ids.dm:920]: I appear to be the stupid, my bad!
Addressing.

Comment (Mothblocks) [reply code/game/objects/items/cards_ids.dm]: You're not stupid

**Review by Mothblocks — APPROVED** [review]

**Pull request merged**